src/Common/src/System/Net/WebSockets/WebSocketValidate.cs

```diff
@@ -123,9 +124,19 @@ internal static void ThrowPlatformNotSupportedException()
 
         internal static void ValidateArraySegment(ArraySegment<byte> arraySegment, string parameterName)
         {
+            Debug.Assert(!string.IsNullOrEmpty(parameterName), "'parameterName' MUST NOT be NULL or string.Empty");
+
             if (arraySegment.Array == null)
             {
-                throw new ArgumentNullException(parameterName + ".Array");
+                throw new ArgumentNullException(parameterName + "." + nameof(arraySegment.Array));
+            }
+            if (arraySegment.Offset < 0 || arraySegment.Offset > arraySegment.Array.Length)
```

Comment:
That is wonderfully sneaky and somewhat scary - do you have a suggestion of what to do with the checks in this PR - should I add them to WebSocketValidate.ValidateArraySegment (could break .net compat) or just add my own validator to HttpWebSocket in HttpListener.

Obviously I'll add tests for any changes